src/order_book_service_order_provider.ts

```diff
@@ -0,0 +1,32 @@
+import { AssetPairsItem, BaseOrderProvider, OrderStore, SignedOrder } from '@0x/asset-swapper';
+
+import { FIRST_PAGE } from './constants';
+import { OrderBookService } from './services/orderbook_service';
+
+const MAX_QUERY_SIZE = 1000;
+// tslint:disable:prefer-function-over-method
+/**
+ * Stubs out the required functions for usage in AssetSwapper/Orderbook
+ * Where required the implementation will fetch from the OrderBookService
+ * representing the underlying database
+ */
+export class OrderBookServiceOrderProvider extends BaseOrderProvider {
+    private readonly _orderbookService: OrderBookService;
+    constructor(orderStore: OrderStore, orderbookService: OrderBookService) {
+        super(orderStore);
+        this._orderbookService = orderbookService;
+    }
+    public async createSubscriptionForAssetPairAsync(_makerAssetData: string, _takerAssetData: string): Promise<void> {
+        return Promise.resolve();
+    }
+    public async getAvailableAssetDatasAsync(): Promise<AssetPairsItem[]> {
+        const response = await this._orderbookService.getAssetPairsAsync(FIRST_PAGE, MAX_QUERY_SIZE);
+        return response.records;
+    }
+    public async destroyAsync(): Promise<void> {
+        return Promise.resolve();
+    }
+    public async addOrdersAsync(_orders: SignedOrder[]): Promise<any> {
+        return Promise.resolve() as any;
+    }
```

Comment:
What are the implications for these stubs? We know they won't get called or be used?

Have you considered making the `OrderBookService` extend `BaseOrderProvider` so we don't have another class.

Or maybe have the orderbook package expose a thinner interface.

Reply:
Some of this is over engineering for sure. It's used in AssetSwapper and in Launch Kit, I can provide a bit of context for the various functions.

| fn | Launch Kit | AssetSwapper | 0x API (swap) |
| --- | ---------- | ----------- | ------ |
| addOrdersAsync | Posts the order to the SRA endpoint | Unused | Unused (No writes happen in swap) |
| createSubscriptionForAssetPairAsync | Subscribes the local storage to any update | Subscribes the local storage to any update | DB is trusted source with writes from another service, no subscription needed |
| getAvailableAssetDatasAsync | Get the assets from the Relayer | Gets the assets from the Relayer | Get the assets from the DB | 
| destroyAsync | Destroy all subscriptions from the above | Destroy subscriptions from the above | No subscriptions were created (could clean up db connection if required) | 
|


The abstraction bbreaks a little because we have the source of truth already and don't require orders to be "Provided" as they always exist in the db.